.github/workflows/test-build-deploy.yml

```diff
@@ -278,91 +281,120 @@ jobs:
       - name: Build Mimir with race-detector
         run: |
           export IMAGE_TAG_RACE=$(make image-tag-race)
-          export MIMIR_IMAGE="grafana/mimir:$IMAGE_TAG_RACE"
+          # We are working on migrating to a distroless image as our only base.
+          # For now we mark our distroless image with a tag prefix `distroless-`
+          # and Alpine remains the default image pushed to the registry.
+          # We always test the distroless image, and we test the alpine image on deploy.
+          export MIMIR_IMAGE="grafana/mimir:distroless-$IMAGE_TAG_RACE"
+          export MIMIR_ALPINE_IMAGE="grafana/mimir:$IMAGE_TAG_RACE"
           make BUILD_IN_CONTAINER=false cmd/mimir/.uptodate_race
-          docker save $MIMIR_IMAGE -o ./mimir_race_image
+          docker save $MIMIR_IMAGE -o ./mimir_race_image_distroless
+          docker save $MIMIR_ALPINE_IMAGE -o ./mimir_race_image
       - name: Upload archive with race-enabled Mimir
         uses: actions/upload-artifact@v4
         with:
           name: Race-enabled Mimir
-          path: ./mimir_race_image
+          path: |
+            ./mimir_race_image
+            ./mimir_race_image_distroless
 
   integration:
-    needs: [goversion, build]
+    needs: [goversion, build, prepare]
     runs-on: ubuntu-latest
     strategy:
       # Do not abort other groups when one fails.
       fail-fast: false
       # Split tests into 6 groups.
       matrix:
+        image_tag_prefix: ['distroless-', '']
         test_group_id:    [0, 1, 2, 3, 4, 5]
         test_group_total: [6]
     steps:
+      - name: Check if we should run against the current matrix.image_tag_prefix
+        id: check_matrix
+        run: |
+          # Run the integration tests against the distroless image tag prefix in all cases.
+          # Only run the integration tests against the (legacy) alpine image when pushing a tag (denoted by no prefix).
+          run_matrix=${{ (needs.prepare.outputs.is_deploy == 'true' || matrix.image_tag_prefix == 'distroless-') && 'true' || 'false'  }}
+          echo "For debugging:"
+          echo "needs.precondition.outputs.isDeploy: ${{ needs.prepare.outputs.is_deploy }}"
+          echo "matrix.image_tag_prefix: ${{ matrix.image_tag_prefix }}"
+          echo "run_matrix: $run_matrix"
+          echo "run_matrix=$run_matrix" >> $GITHUB_OUTPUT
```

Comment:
nit: Can we replace this with github action condition and avoid mixing shell and GHA expressions?

Reply:
Do you have an example of how a condition would work? Do you mean two separate steps using `if` on the steps to then set the output of a variable? Otherwise I am unsure how you can save the output of a github expression.

The alternative woudl be to repeat the use of `needs.prepare.outputs.is_deploy == 'true' || matrix.image_tag_prefix == 'distroless-'` on the following steps

Reply:
I've reworked away from the matrix which clears this up